src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -668,7 +678,7 @@ VIRTUALCommitMemory(
         */
         LPVOID pReservedMemory =
                 VIRTUALReserveMemory( pthrCurrent, lpAddress, dwSize,
-                                      flAllocationType, flProtect );
+                                      flAllocationType | MEM_COMMIT, flProtect, &IsNewMemory );
```

Comment:
The `VIRTUALCommitMemory` is always called with `flAllocationType` containing `MEM_COMMIT`.